### PR TITLE
Added a GPS time listener

### PR DESCRIPTION
**Pull request by jbangelo**
Co-authored-by: benjaminaltieri <benaltieri@swift-nav.com>

### CODE REVIEW

File: src/gps_time_listener.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+const WEEK_SECS: i64 = (WEEK_DAYS * DAY_SECS);
+
+pub struct GpsTimeListener {
+    regex: Regex,
+}
+
+impl GpsTimeListener {
+    pub fn new() -> GpsTimeListener {
+        GpsTimeListener {
+            regex: Regex::new(r"^Time$").unwrap(),
```
Comment (benjaminaltieri) [inline src/gps_time_listener.rs:32]: maybe we can change this to lower case to match `echo`

File: src/gps_time_listener.rs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        }
+    }
+}
+
+impl MessageListener for GpsTimeListener {
+    fn help(&self) -> String {
+        String::from("Converts current time into GPS time. No muss, no fuss.")
```
Comment (benjaminaltieri) [inline src/gps_time_listener.rs:39]: other help strings use backticks, it's pretty nitty but could update as:
```suggestion
        String::from("`time`: Converts current time into GPS time. No muss, no fuss.")
```

**Comment by jbangelo** [comment]
Tested this against the GPS time reported by a Piksi and it appears to be correct (+/- network latency)

File: src/gps_time_listener.rs (hunk 1 above)
Comment (jbangelo) [reply src/gps_time_listener.rs]: Done

File: src/gps_time_listener.rs (hunk 2 above)
Comment (jbangelo) [reply src/gps_time_listener.rs]: Done

**Pull request merged**